app/collections/notes.js

```diff
@@ -1,73 +1,69 @@
-import {
-    saveSuccessful,
-    saveFailed,
-    unableToRetrieve,
-    deletionFailed,
-    updateSuccessful,
-    updateFailed
-} from '../notifications/general';
+import connectionToDB from './connectionSetup';
+import {displayToast} from '../notifications';
 
-const Datastore = require('nedb');
-const electron = require('electron');
-const path = require('path');
+const Notes = connectionToDB('notes');
 
-const userDataPath = (electron.app || electron.remote.app).getPath('userData');
-const collectionsPath = path.join(userDataPath, 'collections');
-const Notes = new Datastore({
-    filename:              path.join(collectionsPath, 'notes.db'),
-    autoload:              true,
-    corruptAlertThreshold: 1,
-    timestampData:         true
-});
+export async function addNewNote(data) {
+    try {
+        await Notes.insert(data);
 
-export const addNewNote = data => {
-    Notes.insert(data, error => {
-        if (error) {
-            saveFailed();
-        }
-        saveSuccessful();
-    });
-};
+        displayToast('saveSuccess');
+    } catch (e) {
+        displayToast('saveFail');
+        console.log(e);
+    }
+}
```

Comment:
@hcuffy I see thousands and thousands of functions with this structure, I guess my previous comment about extracting this makes more and more sense :D

Reply:
I just removed some of them and used instance of instead